packages/leancode_lint/lib/lints/hook_widget_does_not_use_hooks.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/error/error.dart';
+import 'package:analyzer/error/listener.dart';
+import 'package:analyzer/source/source_range.dart';
+import 'package:custom_lint_builder/custom_lint_builder.dart';
+import 'package:leancode_lint/helpers.dart';
+
+/// Displays warning when a `HookWidget` does not use hooks in the build method.
+class HookWidgetDoesNotUseHooks extends DartLintRule {
+  HookWidgetDoesNotUseHooks() : super(code: _getLintCode());
+
+  static const ruleName = 'hook_widget_does_not_use_hooks';
+
+  @override
+  void run(
+    CustomLintResolver resolver,
+    ErrorReporter reporter,
+    CustomLintContext context,
+  ) {
+    context.registry.addClassDeclaration(
+      (node) {
+        final isHookWidget =
+            node.extendsClause?.superclass.name2.lexeme == 'HookWidget';
```

Comment:
Let's never do syntactic checks when a semantic (type level) one is possible. You had previously the `TypeChecker.fromName('HookWidget', packageName: 'flutter_hooks')` check, it should stay

Reply:
isExactly doesn't work